aksparse/server/Aksparse.chpl

```diff
@@ -414,13 +423,35 @@ module Aksparse {
         st.addEntry(result2name, new shared SymEntry(result1_uk1.a));
         var result3name = st.nextName();
         st.addEntry(result3name, new shared SymEntry(result3));
+
+        var gb_row_val = getGroupBy(2, [result1name, result3name], ["pdarray", "pdarray"], false, st);
+        var gb_row_val_uklist = get_unique_keys(gb_row_val, st);
+        var gEnt_gb_row_val_uk0: borrowed GenSymEntry = getGenericTypedArrayEntry(gb_row_val_uklist[0], st);
+        var gb_row_val_symentry = toSymEntry(gEnt_gb_row_val_uk0,int);
+        var gb_row_val_uk = getGroupBy(1, [gb_row_val_symentry.name], ["pdarray"], false, st);
+        var gb_row_val_uk_uklist = get_unique_keys(gb_row_val_uk, st);
+        var gEnt_gb_row_val_uk_uk0: borrowed GenSymEntry = getGenericTypedArrayEntry(gb_row_val_uk_uklist[0], st);
+        var gb_row_val_uk_uk_symentry = toSymEntry(gEnt_gb_row_val_uk_uk0,int);
+
+        var len_arr: [0..0] int = result3.size;
+        var segs2 = concatArrays(gb_row_val_uk.segments.a, len_arr); //Fix this?
+        var diffs2 = segs2[1..segs2.size - 1] - segs2[0..segs2.size - 2];
+        var ind_ptr: [0..self_shape_height] int = 0; //Maybe wrong shape var?
+        ind_ptr[gb_row_val_uk_uk_symentry.a + 1] = diffs2;
```

Comment:
just point out that `ind_ptr` would be a local array here not distributed. And since we'd be doing random writes here, an optimization we could do would be to use copyAggregation. This would be very similar to what is done in `pdarrayIntIndexing` proc above (which i think is actually doing array indexing). The core being this `forall`

```chapel
forall (a1,idx) in zip(aa,iva) with (var agg = newSrcAggregator(int)) {
    agg.copy(a1,a2[idx]); //Performance note
}
```